ttnn/cpp/ttnn/operations/experimental/copy/typecast/typecast.cpp

```diff
@@ -15,14 +15,16 @@ ttnn::Tensor TypecastOperation::invoke(
     uint8_t queue_id,
     const Tensor& input_tensor,
     const DataType& dtype,
-    const std::optional<MemoryConfig>& output_mem_config) {
-    return operation::run(ttnn::operations::data_movement::CopyDeviceOperation{output_mem_config.value_or(input_tensor.memory_config()), dtype}, {input_tensor}, {}, {}, queue_id).at(0);
+    const std::optional<MemoryConfig>& output_mem_config,
+    const std::optional<Tensor>& optional_output_tensor) {
+    return operation::run(ttnn::operations::data_movement::CopyDeviceOperation{output_mem_config.value_or(input_tensor.memory_config()), dtype}, {input_tensor}, {}, {optional_output_tensor}, queue_id).at(0);
 }
 
 ttnn::Tensor TypecastOperation::invoke(
     const Tensor& input_tensor,
     const DataType& dtype,
-    const std::optional<MemoryConfig>& output_mem_config) {
+    const std::optional<MemoryConfig>& output_mem_config,
+    const std::optional<Tensor>& optional_output_tensor) {
```

Comment:
Don’t forget to pass optional output tensor here

Reply:
Done

Reply:
I think it should not be const?

Reply:
The only thing we do with it is return by value in create_output_tensors and there's no direct modification...I'm not actually sure this feature works. Even if I drop the const here the infra actually requires an eventual const for some reason :/. We should probably migrate this to the new infra as part of a bigger refactor as that gets fixed.